apps/src/miniApps/neighborhood/Neighborhood.ts

```diff
@@ -285,4 +290,20 @@ export default class Neighborhood {
       });
     }
   }
+
+  // Returns a promise that resolves when all neighborhood signals have finished processing.
+  waitUntilDone(): Promise<void> {
+    if (this.donePromise) {
+      return this.donePromise;
+    }
+
+    this.donePromise = new Promise(resolve => {
```

Comment:
If this method is called when the neighborhood is not running, what happens?

Reply:
Good question!  Since`this.donePromise` is `null`, a new `Promise` is created and assigned to `donePromise`. Since nothing will trigger `this.resolveOnDone` since signals are not being processed and `DONE` signal won't be reached, the promise will 'hang'.
I'll add a check for `!isRunning` at beginning of method - thanks!